src/webgpu/api/validation/image_copy/buffer_related.spec.ts

```diff
@@ -136,25 +126,24 @@ g.test('bytes_per_row_alignment')
     });
 
     let success = false;
-    // writeTexture doesn't require bytersPerRow to be 256-byte aligned.
+    // writeTexture doesn't require bytesPerRow to be 256-byte aligned.
     if (method === 'WriteTexture') success = true;
     // If the copy height <= 1, bytesPerRow is not required.
     if (copyHeight / info.blockHeight <= 1 && bytesPerRow === undefined) success = true;
     // If bytesPerRow > 0 and it is a multiple of 256, it will succeeed if other parameters are valid.
     if (bytesPerRow !== undefined && bytesPerRow > 0 && bytesPerRow % 256 === 0) success = true;
 
+    const size = [info.blockWidth, !copyHeight ? info.blockHeight : copyHeight, 1];
     const texture = t.device.createTexture({
-      size: [info.blockWidth, !copyHeight ? info.blockHeight : copyHeight, 1],
+      size,
       dimension,
       format,
       usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.COPY_DST,
     });
 
-    t.testBuffer(
-      buffer,
-      texture,
-      { bytesPerRow },
-      { width: info.blockWidth, height: copyHeight, depthOrArrayLayers: 1 },
-      { dataSize: 512 * 8 * 16, method, success }
-    );
+    t.testBuffer(buffer, texture, { bytesPerRow }, size, {
```

Comment:
AFAICT this no longer tests the case where copyHeight is 0. So it's just a duplicate of another case (the one where copyHeight == blockHeight)

Reply:
(Because it was changed to pass `size` here instead of the original `copyHeight`)

Reply:
Agreed! I reset size.height because texture's height cannot be zero, otherwise it will fail to create a texture. However, copyHeight is 0 while texture's height >  0 is not allowed for depth/stencil formats, so I use texture's size as copySize in order to make sure it is a full copy. Then copyHeight == 0 is not useful, like you said. 

In the updated code, I filtered out copyHeight == 0 for depth/stencil formats but use 0 as copyheight for other formats.